File: 1445-number-of-sub-arrays-of-size-k-and-average-greater-than-or-equal-to-threshold/number-of-sub-arrays-of-size-k-and-average-greater-than-or-equal-to-threshold.cpp
class Solution {
public:
    int numOfSubarrays(vector<int>& arr, int k, int threshold) {
        int i=0, j=0, ans = 0, sum = 0;
        while (j<arr.size()){
            sum+=arr[j];
            if (j-i+1 < k){
                j+=1;
            } else {
                if (sum/k >= threshold){
                    ans+=1;
                }
                sum-=arr[i];
                i+=1;
                j+=1;
            }
        }

        return ans;
    }
};

File: 1445-number-of-sub-arrays-of-size-k-and-average-greater-than-or-equal-to-threshold/number-of-sub-arrays-of-size-k-and-average-greater-than-or-equal-to-threshold.go
func numOfSubarrays(arr []int, k int, threshold int) int {
    cnt := 0
    j, sum := 0, 0
    for i := 0; i < len(arr); i++ {
        sum += arr[i]
        win_len := i - j + 1
        if win_len == k {
            average := (sum / win_len)

            if average >= threshold {
                cnt++
            }

            sum -= arr[j]
            j++
        }
    }

    return cnt
}